packages/typegpu/src/resolutionCtx.ts

```diff
@@ -332,6 +332,7 @@ export class ResolutionCtxImpl implements ResolutionCtx {
   readonly #modeStack: ExecState[] = [];
   private readonly _declarations: string[] = [];
   private _varyingLocations: Record<string, number> | undefined;
+  private _currentlyResolvedItems: WeakSet<object> = new WeakSet();
```

Comment:
For new private properties, we can use the recently added ECMAScript standard "private elements".
Can be also made `readonly`, as we never reassign it to anything.
```suggestion
  readonly #currentlyResolvedItems = new WeakSet<object>();
```